Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,254 @@
+#include "Projectiles.h"
+#include "../Helpers.h"
+
+
+Projectile::Projectile(size_t id) : IMovable(id)
+{
+    SetPassable(D_ALL, Passable::AIR);
+    tickSpeed = 0.25;
+    damage = 0;
+    SetSprite("icons/projectiles.dmi");
+    v_level = 5;
+}
+void Projectile::Process()
+{
+    if(set_target_ == 0)
+    {
+       return;
+    }
+    int z = current_step_;
+    Dir step = VDirToDir(movement_[z]);
+
+   if(!TryMove(step))
```

Comment:
No, it can not. But you should not use them - `D_SOUTHEAST` and etc are used only as workarounds for .dmi sprites (for example. some pipes). So do not use diagonal dirs in movement .